src/posix/platform/settings.cpp

```diff
@@ -60,6 +60,13 @@
 
 static ot::Posix::SettingsFile sSettingsFile;
 
+#if OPENTHREAD_POSIX_SETTINGS_PATH_SET_API
+static char sSettingsPath[PATH_MAX] = OPENTHREAD_CONFIG_POSIX_SETTINGS_PATH;
+
+char *ot::Posix::PlatformGetSettingsPath() { return sSettingsPath; }
+void  ot::Posix::PlatformSetSettingsPath(const char *aSettingsPath) { snprintf(sSettingsPath, sizeof(sSettingsPath), "%s", aSettingsPath); }
+#endif
```

Comment:
Could we only wrap the function `PlatformSetSettingsPath` by the  `OPENTHREAD_POSIX_SETTINGS_PATH_SET_API ` ?  So that we can unify the method of getting the settings file path.